File: api/src/main/java/com/api/app/controller/MemberController.java
package com.api.app.controller;

import com.api.app.member.request.LoginRequestDto;
import com.api.app.member.request.SignupRequestDto;
import com.api.app.member.response.TokenResponseDto;
import com.api.app.service.MemberService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/members")
@RequiredArgsConstructor
public class MemberController {

    private final MemberService memberService;

    @PostMapping("/signup")
    public ResponseEntity<Void> signup(@RequestBody SignupRequestDto request) {
        memberService.signup(request);
        return ResponseEntity.status(HttpStatus.CREATED).build();
    }

    @PostMapping("/login")
    public ResponseEntity<TokenResponseDto> login(@RequestBody LoginRequestDto request) {
        TokenResponseDto tokenResponse = memberService.login(request);
        return ResponseEntity.ok(tokenResponse);
    }

    @PostMapping("/refresh")
    public ResponseEntity<TokenResponseDto> refreshToken(@RequestHeader("Refresh-Token") String refreshToken) {
        TokenResponseDto tokenResponse = memberService.refreshToken(refreshToken);
        return ResponseEntity.ok(tokenResponse);
    }
}

File: api/core-service/src/main/kotlin/com/vibepay/core/controller/MemberController.kt
package com.vibepay.core.controller

import com.api.app.common.response.ApiResponse
import com.vibepay.core.dto.request.member.MemberLoginRequest
import com.vibepay.core.dto.request.member.MemberRegisterRequest
import com.vibepay.core.dto.request.member.TokenRefreshRequest
import com.vibepay.core.dto.response.member.MemberInfoResponse
import com.vibepay.core.dto.response.member.MemberLoginResponse
import com.vibepay.core.dto.response.member.TokenRefreshResponse
import com.vibepay.core.service.MemberService
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestHeader
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/members")
class MemberController(private val memberService: MemberService) {

    @PostMapping("/register")
    fun register(@RequestBody @Valid request: MemberRegisterRequest): ApiResponse<Void> {
        memberService.registerMember(request)
        return ApiResponse.success()
    }

    @PostMapping("/login")
    fun login(@RequestBody @Valid request: MemberLoginRequest): ApiResponse<MemberLoginResponse> =
        ApiResponse.success(memberService.login(request))

    @GetMapping("/me")
    fun getMemberInfo(@RequestHeader("X-User-Email") email: String): ApiResponse<MemberInfoResponse> =
        ApiResponse.success(memberService.getMemberInfo(email))

    @PostMapping("/refresh")
    fun refreshToken(@RequestBody @Valid request: TokenRefreshRequest): ApiResponse<TokenRefreshResponse> =
        ApiResponse.success(memberService.refreshToken(request))
}
